### DISCUSSION TITLE
Timezone data example?

### DISCUSSION THREAD

**Discussion by nvcnvn** [valhalla/valhalla#4187] [Q&A] [answered]
Hi, I know this may not a Valhalla specific question but where can I find an example timezone_data?
UPDATE 1:
The time zone I'm asking should be something related to turn restriction with time condition.

**Comment by ImreSamu** [comment, marked as answer]
as I see the Valhalla project using this source:
https://github.com/valhalla/valhalla/blob/e4845b68e8ef8de9eabb359b23bf34c879e21f2b/scripts/valhalla_build_timezones#L24

so you can check the https://github.com/evansiroky/timezone-boundary-builder/ project.

**Comment by nilsnolde** [comment]
Or maybe this: https://github.com/valhalla/valhalla/blob/e4845b68e8ef8de9eabb359b23bf34c879e21f2b/src/baldr/CMakeLists.txt#L5C1-L14. 

The IANA date_time files are being compiled by a custom CMake function into headers which are used by our source code. The data @ImreSamu is referencing is then used to spatially look up those timezones while building the graph.

**Reply by nvcnvn** [reply]
oh thanks @nilsnolde and @ImreSamu 
I should be more specific in my question.
It should be something related to time constraint restriction (for example no u-turn from 16:00 to 20:00).
I'm building a map for a single city so I guess I'm expecting a very small timezone database :D

**Reply by kevinkreiser** [reply]
the database is only used when you build the tiles. the tiles get an integer that represents the timezone where the graph node is located. after building the tiles you can throw the database out the window. we do not support anything other than a full planet wide timezone database it's not that big anyway like a 150mb or something and as I said only temporary

**Reply by nvcnvn** [reply]
OK, got it, thank you @kevinkreiser
